src/utils.jl

```diff
@@ -75,8 +79,12 @@ dim(x::RowVecs) = size(x.X, 2)
 
 pairwise(d::PreMetric, x::RowVecs) = Distances.pairwise(d, x.X; dims=1)
 pairwise(d::PreMetric, x::RowVecs, y::RowVecs) = Distances.pairwise(d, x.X, y.X; dims=1)
-pairwise(d::PreMetric, x::AbstractVector, y::RowVecs) = Distances.pairwise(d, transpose(reduce(hcat, x)), y.X; dims=1)
-pairwise(d::PreMetric, x::RowVecs, y::AbstractVector) = Distances.pairwise(d, x.X, transpose(reduce(hcat, y)); dims=1)
+function pairwise(d::PreMetric, x::AbstractVector, y::RowVecs)
+    return Distances.pairwise(d, transpose(reduce(hcat, x)), y.X; dims=1)
```

Comment:
Just noticed, I guess `permutedims` is more appropriate here since we do not want to apply it recursively:
```suggestion
    return Distances.pairwise(d, permutedims(reduce(hcat, x)), y.X; dims=1)
```